extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -313,6 +339,15 @@ export class AmpA4A extends AMP.BaseElement {
         /** @return {!Promise<?string>} */
         .then(creativeParts => {
           checkStillCurrent(promiseId);
+          // Keep a handle to the creative body so that we can render into
+          // SafeFrame later, if necessary.  TODO(tdrl): Temporary, while we
+          // assess whether this is the right solution to the Safari+iOS iframe
+          // src cache issue.  If we decide to keep a SafeFrame-like solution,
+          // we should restructure the promise chain to pass this info along
+          // more cleanly, without use of an object variable outside the chain.
+          if (creativeParts && creativeParts.creative) {
+            this.creativeBody_ = creativeParts.creative;
```

Comment:
Let's only populate the creativeBody_ field if we know the render method will be 'safeframe' (ensures control flow is as identical to existing flow as possible).  Later you can then remove the 'safeframe' predicate any merely check if creativeBody_ was populated